db-auto-backup.py

```diff
@@ -160,70 +160,154 @@ def get_backup_provider(container_names: Iterable[str]) -> Optional[BackupProvid
 
 
 def get_container_names(container: Container) -> Iterable[str]:
+    """
+    Extract names for a container from image tags or fallback to container name.
+    """
     names = set()
-    for tag in container.image.tags:
-        registry, image = docker.auth.resolve_repository_name(tag)
-        image, tag_name = image.split(":", 1)
-        names.add(image)
+
+    if container.image.tags:
+        for tag in container.image.tags:
+            image_name = tag.split(":")[0].split("@")[0]
+            image_name = image_name.split("/")[-1]
+            names.add(image_name)
+
+    if not names and container.attrs.get("Config", {}).get("Image"):
+        image_name = container.attrs["Config"]["Image"].split(":")[0].split("@")[0]
+        image_name = image_name.split("/")[-1]
+        names.add(image_name)
+
+    if not names and container.name:
+        names.add(container.name)
+
     return names
 
+def is_swarm_mode() -> bool:
+    docker_client = docker.from_env()
+    info = docker_client.info()
+    return info.get("Swarm", {}).get("LocalNodeState") == "active"
 
-@pycron.cron(SCHEDULE)
-def backup(now: datetime) -> None:
-    print("Starting backup...")
 
+def get_local_node_id() -> str:
     docker_client = docker.from_env()
-    containers = docker_client.containers.list()
+    info = docker_client.info()
+    return info["Swarm"]["NodeID"]
 
-    backed_up_containers = []
 
-    print(f"Found {len(containers)} containers.")
+def get_local_node_tasks() -> list:
+    docker_client = docker.from_env()
```

Comment:
Issue: This code creates a lot of extra connections to the docker API - could a client be passed around instead?